coderedcms/models/page_models.py

```diff
@@ -807,20 +807,22 @@ def most_recent_occurrence(self):
             noc = self.next_occurrence()
             if noc:
                 return noc
-            aoc = []
-            for occurrence in self.occurrences.all():
-                aoc += [instance for instance in occurrence.all_occurrences()]
-            if len(aoc) > 0:
-                return aoc[-1]  # last one in the list
 
         except AttributeError:
             # Triggers when a preview is initiated on an
             # EventPage because it uses a FakeQuerySet object.
             # Here we manually compute the next_occurrence
             occurrences = [e.next_occurrence() for e in self.occurrences.all()]
-            if occurrences:
+            # If there are no more occurrences, we find the last one instead
+            if occurrences and None not in occurrences:
                 return sorted(occurrences, key=lambda tup: tup[0])[0]
 
+        aoc = []
```

Comment:
This is the block of code that returns the last occurrence, if both attempts above to find the next occurrence fail, right? If so can you add a comment about it.